options/options.js

```diff
@@ -221,6 +265,81 @@ function loginBitbucket(param) {
   })
 }
 
+function loginGitLabOauth(param) {
+  const username = param.username;
+  const password = param.password;
+  const baseUrl = param.baseUrl;
+  const headers = {}
+  const domain = baseUrl.replace('/api/v4','');
+  $.ajax({
+    url: `${domain}/oauth/token`,
+    headers: headers,
+    method: 'POST',
+    dataType: 'json',
+    crossDomain: true,
+    contentType: 'application/x-www-form-urlencoded',
+    data: {
+      grant_type: 'password',
+      username: username,
+      password: password
+    }
+  })
+    .done(response => {
+      return $.getJSON(
+        `${baseUrl}/user`,
+        { access_token: response.access_token }
+      )
+        .done(user => {
+          chrome.storage.sync.set({scm: param.scm, user: user.username, token: {type : 'oAuth', token :response.access_token}, baseUrl: baseUrl}, () => {
+            location.reload();
+          });
+          chrome.storage.local.get('tab', (item) => {
+            if(item.tab) {
+              chrome.tabs.reload(item.tab);
+            }
+          });
+        });
+    })
+    .fail(err => {
+      if (err.status == 401 &&
+        err.getResponseHeader('X-GitLab-OTP') !== null &&
+        $('.login-item-otp').filter(':visible').length == 0) {
+        $('.login-item').animate({height: 'toggle', opacity: 'toggle'}, 'slow');
+      } else {
+        $('.error').show();
+      }
+    })
+}
+
+function loginGitLabToken(param) {
+  const personalToken = param.personalToken;
+  const baseUrl = param.baseUrl;
+  const headers = {};
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/leonhartX/gas-github/blob/5becb274ac17be98fee6e945e23cbab998e9d302/options/options.js#L317"></span>‘headers’ is assigned a value but never used.</td>
    </tr>
  </tbody>
</table>